packages/expo-background-task/ios/BackgroundTaskAppDelegate.h

```diff
@@ -0,0 +1,13 @@
+// Copyright 2024-present 650 Industries. All rights reserved.
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+#import <ExpoModulesCore/EXSingletonModule.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface BackgroundTaskAppDelegate : EXSingletonModule <UIApplicationDelegate>
```

Comment:
the EXSingletonModule is mostly deprecated. could you share the intention to use here? if the idea is to handle `application:didFinishLaunchingWithOptions:`, you could use an [AppDelegate Subscriber](https://docs.expo.dev/modules/appdelegate-subscribers/#example). then we can also use swift than objc

Reply:
Good point - I'll try to remove all Swift code - also the #ifdef that I moved to Objective-c as you mentioned.